api/yelp_beans/matching/match_utils.py

```diff
@@ -86,3 +88,96 @@ def get_previous_meetings(subscription, cooldown=None):
     disallowed_meetings = {tuple([meeting.id for meeting in meeting]) for meeting in disallowed_meetings}
 
     return disallowed_meetings
+
+
+def jaccard(list1, list2):
+    intersection = len(list(set(list1).intersection(list2)))
+    if intersection == 0:
+        return 1
+    else:
+        union = (len(list1) + len(list2)) - intersection
+        return float(intersection) / union
+
+
+def get_pairwise_distance(
+    user_pair,
+    org_graph,
+    employee_df,
+    max_tenure=1000,
+):
+    """
+    get the distance between two users.
+    The returned distance score is a linear combination of the multiple user attributes' distnace (normalized).
+    The importance of each attribute is considered equal.
+    User attribute considered:
+    1. team/function: distance in the org chart
+    2. location - country, city
+    3. tenure at Yelp
+    4. language
+
+    note: we considered using education and work experience, but think it likely correlates with the first attribute
+    """
+    user_a, user_b = user_pair
+    user_a_attributes = dict(employee_df.loc[user_a])
+    user_b_attributes = dict(employee_df.loc[user_b])
+
+    distance = 0
+    dist_1 = nx.shortest_path_length(org_graph, user_a, user_b)
+    dist_1 = dist_1 / 10  # approx. min-max scaled
+    distance += dist_1
+
+    # location
+    try:
+        user_a_city, user_a_country = user_a_attributes["location"].split(", ")
+    except ValueError:
+        user_a_city, user_a_country = "unkown", user_a_attributes["location"]
+    try:
+        user_b_city, user_b_country = user_b_attributes["location"].split(", ")
+    except ValueError:
+        user_b_city, user_b_country = "unkown", user_b_attributes["location"]
+    country_dist = 0 if user_a_country == user_b_country else 1
+    city_dist = 0 if user_a_city == user_b_city else 1
+    dist_2 = country_dist + city_dist
+    dist_2 = dist_2 / 2  # min-max scaled
+    distance += dist_2
+
+    # tenure
+    dist_3 = abs(int(user_a_attributes["days_since_start"]) - int(user_b_attributes["days_since_start"]))
```

Comment:
Tenure is a bit subjective. I don't have strong opinions here if it doesn't lead to starvation. Fundamental to this assumption is that tenured folks know each other and so optimize for meeting newer less tenured people.

I think this works for v1 but I'll be curious to hear feedback on whether folks not getting matched with similarly tenured people gets noticed. Perhaps eventually we should get to a place where we can ask users to tell us their preferences for matching